qiskit_machine_learning/kernels/algorithms/quantum_kernel_trainer.py

```diff
@@ -0,0 +1,233 @@
+        self._quantum_kernel = quantum_kernel
+
+
+class QuantumKernelTrainer:
+    """
+    Quantum Kernel Trainer.
+    This class provides utility to train ``QuantumKernel`` feature map parameters.
+
+    **Example**
+
+    .. code-block::
+
+        # Create 2-qubit feature map
+        qc = QuantumCircuit(2)
+
+        # Vectors of input and trainable user parameters
+        input_params = ParameterVector("x_par", 2)
+        user_params = ParameterVector("θ_par", 2)
+
+        # Create an initial rotation layer of trainable parameters
+        for i, param in enumerate(user_params):
+            qc.ry(param, qc.qubits[i])
+
+        # Create a rotation layer of input parameters
+        for i, param in enumerate(input_params):
+            qc.rz(param, qc.qubits[i])
+
+        quant_kernel = QuantumKernel(
+            feature_map=qc,
+            user_parameters=user_params,
+            quantum_instance=...
+        )
+
+        loss_func = ...
+        optimizer = ...
+        initial_point = ...
+
+        qk_trainer = QuantumKernelTrainer(
+                                        quantum_kernel=quant_kernel,
+                                        loss=loss_func,
+                                        optimizer=optimizer,
+                                        initial_point=initial_point,
+                                        )
+        qkt_results = qk_trainer.fit(X_train, y_train)
+        optimized_kernel = qkt_results.quantum_kernel
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: QuantumKernel,
+        loss: Union[str, KernelLoss] = "svc_loss",
+        optimizer: Optional[Optimizer] = None,
+        initial_point: Optional[Sequence[float]] = None,
+    ):
+        """
+        Args:
+            quantum_kernel: QuantumKernel to be trained
+            loss (str or KernelLoss): Loss functions available via string:
+                       {'svc_loss': SVCLoss()). If a string is passed as the
+                       loss function, then the underlying KernelLoss object
+                       will exhibit default behavior.
+            optimizer: An instance of ``Optimizer`` to be used in training. Since no
+                       analytical gradient is defined for kernel loss functions, gradient-based
+                       optimizers are not recommended for training kernels.
+            initial_point: Initial point from which the optimizer will begin.
+
+        Raises:
+            ValueError: unknown loss function
+        """
+        # Class fields
+        self._quantum_kernel = quantum_kernel
+        self._initial_point = initial_point
+        self._optimizer = optimizer or SPSA()
+
+        # Set up loss function
+        if isinstance(loss, str):
+            loss = loss.lower()
+            if loss == "svc_loss":
+                self._loss = self._str_to_loss(loss)
+            else:
+                raise ValueError(f"Unknown loss {loss}!")
```

Comment:
I guess a string to loss conversion is done in the `_str_to_loss`, so these checks are unnecessary as you mentioned below.